src/lib/math.js

```diff
@@ -361,8 +361,42 @@ const $builtinmodule = function (name) {
         throw new Sk.builtin.NotImplementedError("math.perm() is not yet implemented in Skulpt");
     };
 
-    function prod(x) {
-        throw new Sk.builtin.NotImplementedError("math.prod() is not yet implemented in Skulpt");
+    function prod_float(iterable, start) {
+        iterable = Sk.abstr.iter(iterable);
+        let result = Sk.ffi.remapToJs(start);
+        for (let x = iterable.tp$iternext(); x !== undefined; x = iterable.tp$iternext()) {
+            if (!Sk.builtin.checkNumber(x)) {
+                throw new Sk.builtin.TypeError("must be a number in prod_float");
+            }
+            result = result * Sk.ffi.remapToJs(x);
+        }
+        return result;
+    }
+
+    function prod(iterable, start) {
+        if (!Sk.builtin.checkIterable(iterable)) {
+            throw new Sk.builtin.TypeError("'" + Sk.abstr.typeName(iterable) + "' object is not iterable");
+        }
+        if (Sk.builtin.checkFloat(start)) {
+            return new Sk.builtin.float_(prod_float(iterable, start));
+        }
+        if (!Sk.builtin.checkInt(start)) {
+            throw new Sk.builtin.TypeError("start must be a number");
+        }
+
+        let result = Sk.ffi.remapToJs(start);
+        result = JSBI.BigInt(result);
+        prod_iterable = Sk.abstr.iter(iterable);
+        for (let x = prod_iterable.tp$iternext(); x !== undefined; x = prod_iterable.tp$iternext()) {
+            if (!Sk.builtin.checkNumber(x)) {
+                throw new Sk.builtin.TypeError("must be a number in prod function");
+            }
+            if (!Sk.builtin.checkInt(x)) {
+                return new Sk.builtin.float_(prod_float(iterable, start));
```

Comment:
This looks like we might end up doing two iterations and creating a second iterable.